client/app/components/queries/visualization-embed.js

```diff
@@ -38,7 +38,8 @@ export default function init(ngModule) {
     return session($http, $route, Auth).then(() => {
       const queryId = $route.current.params.queryId;
       const query = $http.get(`api/queries/${queryId}`).then(response => response.data);
-      const queryResult = $http.post(`api/queries/${queryId}/results`, { parameters: queryStringParameters() }).then(response => response.data);
+      // eslint-disable-next-line no-unused-vars
+      const queryResult = $http.post(`api/queries/${queryId}/results`, { parameters: queryStringParameters() }).then(response => response.data, _ => ({}));
```

Comment:
A polite way to solve this would have been to instantiate a `Query` right here, and see if all parameters are supplied, but since this is happening before Query is injected, I'm not sure if that's even possible.
So basically I'm ignoring the error, knowing that in the case of an error, the same request will be launched from `QueryResult` later on.

Reply:
Why not load the results after we have the query? 

It used to work this way, because we wanted to load everything and then render, assuming there will always be something to load. Now that we're making embed more interactive, it makes sense to just show a loading UI and load things properly.

Reply:
That's good context. Thanks, I'll modify this.

Reply:
I'm thinking it might be worth considering to keep it _as is_ - loading the results after we have the query will result in 2 consecutive roundtrips for _all_ embed page loads. If we keep the change suggested in this PR, it'll result in an extra roundtrip only for the rare case a parameter is missing. WDYT?